packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -545,6 +544,11 @@ export const useGridColumnResize = (
   });
 
   const stopListening = React.useCallback(() => {
+    // Resize didn't start;
+    if (!columnHeaderElementRef.current) {
+      return;
+    }
```

Comment:
We already run this logic lower, so we might as well skip everything if the reference is null.